### ISSUE TITLE
What is a correct .cosmo file style

### ISSUE THREAD

**Issue by GangqiangYu** [usnistgov/COSMOSAC#14]
I obtained the .cosmo file (e.g., ethanol.cosmo as shown the attachment) by performing the Dmol3 caculation in Materials Studio package, but this file style is different from that provided at https://github.com/usnistgov/COSMOSAC (e.g., LFQSCWFLJHTTHZ-UHFFFAOYSA-N.cosmo as shown the attachment). How do I get the same files as yours, and what post-processing is required?
 I look forward to hearing from you！
[ethanol.zip](https://github.com/usnistgov/COSMOSAC/files/6544871/ethanol.zip)

**Comment by ianhbell** [comment]
Yes, it seems that Dmol3 has changed their cosmo file format.  Interested in updating the parser?

**Comment by GangqiangYu** [comment]
It is failed to obtain a sigma profile by running the python script based on the  cosmo file with changed style , but it is successful for your cosmo file. Thank you very much for your help! I am very interested in updating the parser to obtain sigma files successfully.

**Comment by ianhbell** [comment]
We might want to be a bit thoughtful about trying to support all the different cosmo formats that have been generated by Dmol3.  Can you do a bit of research to track down the documentation for their different formats?

**Comment by GangqiangYu** [comment]
I will produce some cosmo files by using the Dmol3 calculation, and then send to you. What do you think ?

**Comment by ianhbell** [comment]
I'm looking for something more conclusive: documentation from Dmol3 staff about all changes that have happened to this file format over the years so that it is clear that we are covering all the relevant formats.

**Comment by ianhbell** [comment]
Ask the support team of Dmol3 for a complete changelog of their cosmo file format.

**Comment by mabounassif** [comment]
Is there a plan to support nwchem COSMO files? or other rather open source alternatives?

**Comment by mabounassif** [comment]
Would be happy to help out with the parser for that bit.

**Comment by ianhbell** [comment]
Do you have an example you can share?

**Comment by mabounassif** [comment]
I'm still learning the ropes on how to effectively use COSMO with nwchem. I ran the following nw file that I've taken from one of their tutorials:
```
start  
echo  
 title "h2o"  
geometry  
 o                  .0000000000         .0000000000        -.0486020332  
 h                  .7545655371         .0000000000         .5243010666  
 h                 -.7545655371         .0000000000         .5243010666  
end  
basis
 o library 6-31g**  
 h library 6-31g**  
end  
cosmo  
 dielec 78.0  
 radius 1.40  
        1.16  
        1.16  
 lineq  0  
end  
task scf energy
```

The output is attached. Let me know if it makes sense
[h2o.txt](https://github.com/usnistgov/COSMOSAC/files/8105442/h2o.txt)
.

**Comment by ianhbell** [comment]
Not clear to me, can you extract the segment charge densities on the segments? I don't see that.

**Comment by ianhbell** [comment]
Well, if you can figure out how to do it, I would be very interested.  It would be nice to have a fully-open implementation from start to finish.

**Comment by mabounassif** [comment]
I've found that a xyz file is generate that is the charge density of the surface. Attached is an example. I'm not too knowledgeable but couldn't we use this directly to calculate the sigma profile? Instead of using segments?
[cosmo.xyz.txt](https://github.com/usnistgov/COSMOSAC/files/8112005/cosmo.xyz.txt)

**Comment by ianhbell** [comment]
I dunno. Can you ask on their forums? We need something like looks like one of the cosmo files: https://github.com/usnistgov/COSMOSAC/blob/master/profiles/UD/cosmo/ACKALUBLCWJVNB-UHFFFAOYSA-N.cosmo

**Comment by mabounassif** [comment]
After a lot of head banging, I've decided to use mopac.
[example.cos.txt](https://github.com/usnistgov/COSMOSAC/files/8120187/example.cos.txt)

**Comment by ianhbell** [comment]
So now you need a parser for that file, and to re-parameterize all of COSMO-SAC. Good luck!

**Comment by ianhbell** [comment]
I was a little bit joking in my previous comment, but that is actually what is required.

**Comment by tcxiaoma** [comment]
Hi, I also encountered the same problem. What version of mod3(MS) should match the program?

**Comment by EmiliaSX** [comment]
Hello,

I have tried three versions of Material Studio (MS7.0, MS2018, and MS2020), all using the DMol3 module. However, the .cosmo output files generated from these versions are different from those in the COSMOSAC UD database. Could you please advise me on which version of Material Studio I should use to meet the requirements of the to_sigma.py script?

Thank you for your help!
[MS7-2020test.zip](https://github.com/user-attachments/files/17842458/MS7-2020test.zip)

**Comment by ianhbell** [comment]
I'm not sure. You might want to consider to write a new parser to handle this new format and submit it as a pull request.

**Comment by EmiliaSX** [comment]
> I'm not sure. You might want to consider to write a new parser to handle this new format and submit it as a pull request.

Thank you for your response. I have manually adjusted the output from MS into the input format required by COSMOSAC, and the calculations are proceeding smoothly. As a next step, I am considering writing a code to automate this process.

Now, I would like to inquire about your program's capability regarding ionic liquids or deep eutectic solvents. Many researchers split these into cations and anions, or hydrogen bond donors/acceptors, and perform COSMO calculations separately, subsequently combining the cavity volumes and Sigma profiles in a weighted manner. Is your program capable of handling such operations? (From my understanding, the VT-2005 database, which uses COSMO-SAC2002, supports this approach as it only requires the cavity volume and sigma profiles. However, I'm uncertain about the UD databases, such as COSMO-SAC2010 and COSMO-SAC-DSP models. Are these compatible?)

**Comment by ianhbell** [comment]
For other readers, could you write up what changes were required?

I am not sure about that, I know there was another researcher asking along these lines recently (#31). Perhaps you could connect with them.
